### ISSUE TITLE
First click after starting the app doesn't start the station

### ISSUE THREAD

**Issue by mohamedsakhri** [haecker-felix/gradio#69]
After starting the app when you click on a station the miniplayer appears but it doesn't start playing. You should click a second time to play the station.
This behaviour happens only after you start the application.
I think this is a bug.

**Comment by haecker-felix** [comment]
Uhmm the mini player should be disabled because it is too buggy. Do you have the mini player icon in the headerbar?

**Comment by mohamedsakhri** [comment]
In the headerbar? Can you attach a screenshot showing what you mean..

Well the behaviour is not only about mini player.. just follow this steps:
1. start the app
2. click on a station ( =>it doesn't start playing)
3. click on it again ( => it starts playing)

**Comment by kajzersoze** [comment]
I can confirm this bug.
Takes two clicks on a station to start playing, must be something in latest commits because it used to work with one click few days ago.
btw I don't know what headerbar is nor mini player.

**Comment by mohamedsakhri** [comment]
I don't know what is headerbar neither! 
Miniplayer, as I understand, is the station's player in the bottom. It is marked in the screenshot..

![gradio_miniplayer](https://cloud.githubusercontent.com/assets/2544216/18180472/449d4a94-7087-11e6-9206-50458274c646.png)

**Comment by kajzersoze** [comment]
What is then main player, I mean non-MiniPlayer , or the other player (whatever it's called) beside mini player ?

**Comment by mohamedsakhri** [comment]
I'm not sure but I think it's the one involved when you click on a station, also the station's item itself. In the "old" design it was as a popup..

@haecker-felix  Should now better :)

**Comment by kajzersoze** [comment]
I liked the old design better :)

**Comment by mohamedsakhri** [comment]
I'm sure there is much to improve in the design. Maybe a brainstorming topic about it can help collecting some ideas..

**Comment by kajzersoze** [comment]
At least he can add tooltips for each button which would describe what each button would do, took me a long time to figure out how to add stations and remove them in this new design, that button on the bottom with star icon on it was doing that, but somehow I wasn't expecting that, used to be + / - button in the old design.

It's not that this new design is bad, but it's very confusing.
I told him in other topic how to add tooltips for one button, I can do that for him for the whole app and make a commit, but why bother if he decides all of the sudden to remove it and go with something completely new :)

He's totally unpredictable, for example I can fix this double click bug but I don't even look at the code anymore precisely because of that reason, he might re-design the whole thing tomorrow, so there's no point :)

**Comment by mohamedsakhri** [comment]
I see what you are talking about. It's basically a one person project, so the lack of planning  on a long term and unpredictability are maybe okay, at least for him.. 

A roadmap for the next steps, more documentation would be however very helpful to get more people involved :)

**Comment by kajzersoze** [comment]
Agreed.

**Comment by haecker-felix** [comment]
Good morning. I was a bit confused, because i named a other widget "mini player". This what you mean that is the player toolbar. Now is the bug a bit more clear to me. The headerbar is the top bar with discover/library.

Yeah the old design was better in some points, but the New design isn't finished and i have many ideas to improve it (like right click menu).

I'll add tooltips! Sorry I haven't time yet. Since yesterday i started my new full time shop, so i'm a bit busy now.

Yeah i admit i often completely redesign things. Sorry :/. The 4.1 release should squash many bugs and improve the data backend. The current backend is horrible and buggy, so i rewrite many things within the time.

The documentation is also missing... English isn't my main language, so it is a bit hard for me too write longer texts.

@kajzersoze if you want add new features / fix bugs, feel free to ask me. Than i can say you if i'm currently rewriting the whole code base or not :). The situation should be better with the 5.0 release.

**Comment by mohamedsakhri** [comment]
You could write the documentation in your main language. I think it's German, isn't it? There will be people who can help translate it or at least use a translator to translate it for themselves if they want to get involved...

**Comment by haecker-felix** [comment]
sorry for delayed answer. 
I think the best idea for me is, making more comments in the code, which explains what the function does.

**Comment by haecker-felix** [comment]
Mhm. Cannot reproduce. Happen this on all stations?

**Comment by mohamedsakhri** [comment]
Yes.
It happens only the first time you click on a station after you start the application.

**Comment by haecker-felix** [comment]
Could you try it with the latest commit? Thanks.

**Comment by kajzersoze** [comment]
For me it happened not only when you first start application, but even later, I just tested it with latest commits and it's still like that. For example I play one radio station then later I click on some other station and I can hear the sound from OS when something is clicked, so it's not like I somehow missed to click it, but still it doesn't play, I need to click one more time on it and then it plays. And it's not always like that, sometimes it needs 2 clicks, sometimes it needs just one.
But that's not a big problem, with this latest commits some stations are not displaying notifications.
In fact, most of them are not showing notifications while I can see a station name and a song title displayed on labels. Also the whole app crashes when I click on a "Infowars" station, you can hear it playing for a few seconds then app just crashes, tried several times and it crashed every time.

**Comment by kajzersoze** [comment]
Ah, never mind about crashing and notifications not showing, it was because of my additional patches, I'll look at that later.

But without any patches, it still needs 2 clicks to start playing (not always)

**Comment by haecker-felix** [comment]
Mhm. Is this station specific?

And what does your patches do (except libnotify)?

**Comment by mohamedsakhri** [comment]
According to my tests this bug is fixed.

- The place of the new sidebar is better than the previous one. I think it would be better if you make its background different than the other parts of the window, maybe a little bit darker. Take a look at [Clementine](https://www.clementine-player.org/), it has a similar sidebar.

- When you make a search and then after that you click the discover button it shows nothing. I think it's a bug.

- Now that you have the sidebar on the left, I think it would be better to add Library and Discover to the sidebar as two new items and leave the headbar as a large search box.

**Comment by mohamedsakhri** [comment]
I think you should start a new general topic/issue about the design if you want to discuss a change or have feedback because feedback here will be lost after the issue is closed.

**Comment by kajzersoze** [comment]
Nah, it wasn't libnotify, I just added icon support for gnotification, here's the patch :
http://www.hastebin.com/nexipoluru.avrasm

I found the problem, it was App.player.tag_homepage but I removed that part, I'll deal with that later.

**Comment by haecker-felix** [comment]
@kajzersoze ah okay. This part will be useful for the new image class :+1: Are there specific stations which have this bug?

@mohamedsakhri 
1) Good idea. I will test a other color.
2) Could you be more specific? I cannot reproduce..
3) I dont think so. Because the sidebar is "discover" related. I think if i would add all items to the sidebar it would be confusing (like clicking on "languages". Does it show items from the library or globally?). But i will add a sidebar to the library tab soon. (actually it is nonsense. no items to add)
4) Agree, but where? Opening a new issue is a bit wrong.

**Comment by kajzersoze** [comment]
You mean bug with single / double click ?
If yes then answer is no, it happens with all the stations, there's no rule why and when that happens.
But that doesn't bother me at all, I was just confirming the bug, you can close it if you want.

As for notifcations and crashing issue I had earlier, it was because of this :
```
		private void send_notification(string summary, string body){
			if (App.player.tag_homepage != "" && App.player.tag_title != ChannelCurrentTitleLabel.get_text())
				Util.send_notification(summary, body, App.player.tag_homepage);
			else
				Util.send_notification(summary, body, station.Icon);
		}
```

that worked until today, so for now I just left : Util.send_notification(summary, body, station.Icon);

**Comment by mohamedsakhri** [comment]
2) You can reproduce it following this user case:

1. enter something in the search box and click the search icon or enter key
2. the search results are displayed
3. remove the search item from the search box
4. when you click discover the default lists ( most popular, recently played..) should be displayed, but nothing is displayed (see Screenshot)

The discover button should show the default lists when the search box is empty or inactive.

![gradio_001](https://cloud.githubusercontent.com/assets/2544216/18419658/2fde3786-7860-11e6-85fb-5ab180361d32.png)

**Comment by kajzersoze** [comment]
Ah yes, you removed EndsWithFoo function in "new sidebar in the discover tab" commit.
That's why App.player.tag_homepage was crashing, okay.

**Comment by mohamedsakhri** [comment]
@haecker-felix  4) What about adding "submit feedback..." in the App's menu where you have "Settings, report an error, about, quit"? The item can than link to a general issue that can stay open for feedback (not errors) or to a wiki entry or  to send an email.

**Comment by haecker-felix** [comment]
But itself: I cannot reproduce it in the latest commit. If the bug still persist, please reopen.

Yeah working on a new feedback option ;)
